lib/cdo/email_preference_constants.rb

```diff
@@ -0,0 +1,12 @@
+module EmailPreferenceConstants
+  SOURCE_TYPES = [
```

Comment:
Should I define this as a hash instead?
``` ruby
SOURCE_TYPES = {
     ACCOUNT_SIGN_UP => 'Teacher account sign up',
     ACCOUNT_TYPE_CHANGE => 'Student account change to teacher',
     ACCOUNT_EMAIL_ADD => 'Add email to teacher account',
     ACCOUNT_EMAIL_CHANGE => 'Update teacher account email',
     FORM_HOUR_OF_CODE => 'Host Hour of Code form',
     FORM_VOLUNTEER => 'Volunteer form',
     FORM_ACCESS_REPORT => 'Access Report form',
     FORM_REGIONAL_PARTNER => 'Regional partner contact'
   }.freeze
```

I think usage would be more precise?  `EmailPreferenceConstants::SOURCE_TYPES::FORM_ACCESS_REPORT`?

Reply:
That won't work as written, because the constants still need to be defined somewhere first. You could use symbols as they keys:
```ruby
SOURCE_TYPES = {
  account_sign_up: 'Teacher account sign up',
  # ...
}

# references by
SOURCE_TYPES[:account_sign_up]
```

or if you want it to be more JavaScript-y, you could use OpenStruct:
```ruby
SOURCE_TYPES = OpenStruct.new(
  account_sign_up: 'Teacher account sign up',
  #...
)

# referenced by 
SOURCE_TYPES.account_sign_up
```